pkg/controller/replicaset/replica_set.go

```diff
@@ -707,22 +709,27 @@ func (rsc *ReplicaSetController) syncReplicaSet(ctx context.Context, key string)
 	if err != nil {
 		return err
 	}
-	// Ignore inactive pods.
-	filteredPods := controller.FilterActivePods(logger, allPods)
 
-	// NOTE: filteredPods are pointing to objects from cache - if you need to
+	// NOTE: activePods and terminatingPods are pointing to objects from cache - if you need to
 	// modify them, you need to copy it first.
-	filteredPods, err = rsc.claimPods(ctx, rs, selector, filteredPods)
+	var terminatingPods []*v1.Pod
```

Comment:
nit: move this down to the `if utilfeature.DefaultFeatureGate.Enabled(features.DeploymentPodReplacementPolicy) {` block to make it clearer that the active pods calculation doesn't care about this variable